File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.cpp
#include <iostream>

using namespace std;

int solution(int n, int w, int num) {
  int row = (num - 1) / w; // 층수 0부터 시작
  int col;
  if (row % 2 == 0) {
    col = (num - 1) % w; // 짝수층은 왼쪽에서 오른쪽으로
  } else {
    col = w - 1 - (num - 1) % w; // 홀수층은 오른쪽에서 왼쪽으로
  }

  cout << "row: " << row << ", col: " << col << endl;

  int answer = 0;

  for (int i = row; i <= (n - 1) / w; i++) {
    int boxNum;
    if (i % 2 == 0) {
      boxNum = i * w + (col + 1);
    } else {
      boxNum = i * w + (w - col);
    }
    if (boxNum > n) {
      break;
    }
    answer++;
    cout << "boxNum: " << boxNum << endl;
  }

  return answer;
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n: int, w: int, num: int) -> int:
    row = (num - 1) // w
    if row % 2 == 0:
        col = (num - 1) % w
    else:
        col = w - 1 - (num - 1) % w
    answer = 0
    max_row = (n - 1) // w
    for current_row in range(row, max_row + 1):
        if current_row % 2 == 0:
            box_num = current_row * w + (col + 1)
        else:
            box_num = current_row * w + (w - col)
        if box_num > n:
            break
        answer += 1
    return answer
